### DISCUSSION TITLE
清理链接

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#2734] [插件说明]
描述
---

> 去掉某些网站的中转链接以及正文的全部链接。

地址
---

[https://simpread.ksria.cn/plugins/details/1VIRZ7VGc3](https://simpread.ksria.cn/plugins/details/1VIRZ7VGc3)

选项
---

随便进入一个页面 e.g. https://sspai.com/post/71576 并进入 **阅读模式 → 右下角 → 动作 → 插件触发器**

按照下图所示即可查看此插件的选项页。（注意：如果你的界面没有插件触发器，请根据此 [解决方案](https://github.com/Kenshin/simpread/discussions/2342) 修复。

<img width="1053" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/6eb39b0f-216b-443f-b99c-71784c57df08">

功能
---

- 适配站点，包括： `简书` `知乎` `掘金` `少数派`

   > 如果你有需要去掉中转链接的网址，请 [提交到这里](https://github.com/Kenshin/simpread/discussions/2734#discussioncomment-1246914)。

- 打开超链接时的方式，包括： `默认方案` `在新的页面打开` `在当前页面打开`

   <img width="455" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/96de21bc-b1f8-4552-a06e-f69dc97775e2">

- [清理正文全部超链接](https://github.com/Kenshin/simpread/discussions/2734#discussioncomment-6371147)

    ![image](https://github.com/Kenshin/simpread/assets/81074/226246a5-38e8-4920-b222-49f0cf3dffe7)

安装与升级
---

- 没有【使用同步助手 · 自动同步】的话，可通过[此链接](https://simpread.ksria.cn/plugins/details/1VIRZ7VGc3) 安装。

- 已经【使用同步助手 · 自动同步】的话，请使用 [插件管理器](https://www.yuque.com/kenshin/simpread/fcw60u#ymS1C) 安装。

更新日志
---

### 1.0.3

- `Fix` 白名单为空时的判断错误

### 1.0.2

- `Add` 加入 `sspai.com`

- `Add` [清理正文全部超链接](https://github.com/Kenshin/simpread/discussions/2734#discussioncomment-6371147) 

   > 关联 Issues [#4799](https://github.com/Kenshin/simpread/issues/4799) [#5674](https://github.com/Kenshin/simpread/issues/5674) 

### 1.0.1

- `Optimize` 为配合 [插件管理器](https://github.com/Kenshin/simpread/discussions/4221) 的更新方式做了数据结构的优化。

### 1.0.0

- 适配站点，包括： `简书` `知乎` `掘金`

- 设置
    - 点击链接后的打开方式，包括： `默认方案` `在新的页面打开` `在当前页面打开`

**Comment by Kenshin** [comment]
> 请在此提交你的需要清理的链接

**Reply by Windy147** [reply]
请问，可否让用户自定义清理文本中的链接？因为标注的时候总会跳转，很不方便。

**Reply by Kenshin** [reply]
> 请问，可否让用户自定义清理文本中的链接？因为标注的时候总会跳转，很不方便。

嗨，可以详细说说你的需求。

**Reply by Windy147** [reply]
<img width="947" alt="2021-11-30 (2)" src="https://user-images.githubusercontent.com/57313069/143983724-0e5a8da6-0b0e-48fb-95c0-4096eab37070.png">



谢谢。

未加超链接的文本可以直接选中标注，而有超链接的文本，不能 “直接” 选中标注，页面会显示跳转的链接。这样对我来说会很不方便。

**Reply by Kenshin** [reply]
> 而有超链接的文本，不能 “直接” 选中标注

你可以从旁边选择它，看下图的视频操作

https://user-images.githubusercontent.com/81074/143985716-7c8c6335-d3dd-4379-bc66-5b55127f08b2.mp4

**Comment by fengqiii** [comment]
请问可以去除wiki上的链接吗，链接特别多，导出到本地的logseq之后很难看，不过要是能改成[[]]包裹的就很棒了。现在可以设置吗？

**Reply by Kenshin** [reply]
你的意思是将维基百科正文内的全部链接：

1. 去掉超链接
2. 改为 `[[xxxx]]` 形式

**Reply by navyningyue** [reply]
对的 wiki的太碍眼了

**Reply by Kenshin** [reply]
嗨，各位。

此插件升级到 1.0.2 版，增加了 [清理正文全部超链接](https://github.com/Kenshin/simpread/discussions/2734#discussioncomment-6371147) 功能。

**Comment by Kenshin** [comment]
# 清理正文全部超链接

某些网页的正文会加入很多不必要的超链接 e.g. 维基百科 或 [知乎问答页 ](https://github.com/Kenshin/simpread/issues/4799) 而这个功能就是为了清理这些页面正文的超链接而设置的。

<img width="457" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/f607b9b5-267a-427d-9181-61f3e2118d20">

使用后会将正文的全部超链接删除。

## 用法

- 全 URL e.g. `https://www.zhihu.com/question/524614207/answer/2412604403`

- 主域名 e.g. `wikipedia.org` （会自动忽略 `www` ）

一个例子

```
wikipedia.org
https://www.zhihu.com/question/524614207/answer/2412604403
```

## 注意

因为这个功能比较特殊，所以 **仅支持白名单模式，不支持自动删除任意页的超链接**。

**Comment by anrgct** [comment]
现在知乎的链接变成这样的[刘铁男](https://www.zhihu.com/search?q=%E5%88%98%E9%93%81%E7%94%B7&search_source=Entity&hybrid_search_source=Entity&hybrid_search_extra=%7B%22sourceType%22%3A%22answer%22%2C%22sourceId%22%3A40499738%7D)，能不能加个正则匹配，只过滤`https://www.zhihu.com/search`这个URL下面的

**Reply by Kenshin** [reply]
1. 暂时不考虑复杂场景
2. 我没太理解你的问题...

**Reply by anrgct** [reply]
我的诉求很好理解啊，我希望去除知乎回答正文中所有自动添加的超链接，保留作者、文章的超链接，通过添加`zhihu.com`规则，所有的知乎超链接都没了，所以希望有一个更细致的正则规则，当然目前也勉强够用

**Reply by Kenshin** [reply]
> 诉求很好理解啊，我希望去除知乎回答正文中所有自动添加的超链接，保留作者、文章的超链接，通过添加`zhihu.com`规则，所有的知乎超链接都没了，所以希望有一个更细致的正则规则，当然目前也勉强够用

okay 我 Get 到了，更新后通知你。

**Reply by anrgct** [reply]
感谢大佬，你只要记录一下就可以了，我也不要求处理，毕竟勉强够用

**Reply by Kenshin** [reply]
小事~ 你说的对，并不是每个知乎里面的链接需要清理的~
